File: examples/csharp/QueueClient.cs
using System.Threading.Channels;

namespace LocoAware.Webhook;

/// <summary>
/// Stand-in queue client. Backed by an in-memory <see cref="Channel{T}"/>.
///
/// TODO: replace with your real broker:
///   - AWS SQS / SNS        AWSSDK.SQS, AWSSDK.SNS
///   - Azure Service Bus    Azure.Messaging.ServiceBus
///   - Apache Kafka         Confluent.Kafka
///   - RabbitMQ             RabbitMQ.Client
///   - Google Pub/Sub       Google.Cloud.PubSub.V1
///   - Or an `inbox` table drained by a <see cref="BackgroundService"/>.
///
/// <see cref="PublishAsync"/> must return only once the message is durably
/// persisted. Do not ack the HTTP request before this completes.
/// </summary>
public class QueueClient
{
    public record Envelope(string Topic, byte[] Body);

    private readonly Channel<Envelope> _channel = Channel.CreateUnbounded<Envelope>();

    public ChannelReader<Envelope> Reader => _channel.Reader;

    public virtual ValueTask PublishAsync(string topic, byte[] body)
        => _channel.Writer.WriteAsync(new Envelope(topic, body));
}

File: examples/java/src/main/java/com/example/locoaware/QueueClient.java
package com.example.locoaware;

import com.fasterxml.jackson.databind.JsonNode;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

/**
 * Stand-in queue client.
 *
 * TODO: swap this for a real broker. Options:
 *   - AWS SQS / SNS      spring-cloud-aws-messaging
 *   - Apache Kafka       spring-kafka
 *   - RabbitMQ           spring-rabbit
 *   - Google Pub/Sub     spring-cloud-gcp-pubsub
 *   - Azure Service Bus  azure-messaging-servicebus
 *   - Or an `inbox` table drained by a scheduled worker.
 *
 * The controller only calls {@link #publish(String, JsonNode)} and expects it
 * to return once the message is durably persisted. Do not ack the HTTP request
 * before this returns — if the JVM dies in between you lose the message.
 */
@Component
public class QueueClient {

    private final ApplicationEventPublisher bus;

    public QueueClient(ApplicationEventPublisher bus) {
        this.bus = bus;
    }

    public void publish(String topic, JsonNode message) {
        bus.publishEvent(new QueueMessage(topic, message));
    }

    public record QueueMessage(String topic, JsonNode body) {}
}
